architecture-decisions/0003-preservation.md

```diff
@@ -0,0 +1,116 @@
+# 3. Preservation
+
+Date: 2019-04-02
+
+## Status
+
+Accepted
+
+## Context
+
+We have agreed that we will preserve digital objects by saving resources in
+Google Cloud Storage in a directory structure which preserves both the binaries
+the resource is made up of as well as the JSON serialization of the resource
+itself.
+
+## Decisions
+
+1. Preserving
+   1. We will preserve materials in Google Cloud Coldline Storage with
+      `versioning` enabled. Versions will be kept indefinitely and without
+      limit. All files will go in a single bucket.
+      - Staging bucket is configured with the following command:
+        ```
+        gsutil mb -c regional -l us-west1 -p pulibrary-figgy-storage-1 gs://figgy-staging-preservation
+        echo '{"rule": [{"action": {"type": "Delete"}, "condition": {"age": 2}}]}' > lifecycle.json
+        gsutil lifecycle set lifecycle.json gs://figgy-staging-preservation
+        rm lifecycle.json
+        gsutil bucketpolicyonly set on gs://figgy-staging-preservation
+        gsutil iam ch serviceAccount:figgy-staging@pulibrary-figgy-storage-1.iam.gserviceaccount.com:objectAdmin gs://figgy-staging-preservation
+        ```
+      - Production bucket is configured with the following command:
+        ```
+        gsutil mb -c coldline -l us-west1 -p pulibrary-figgy-storage-1 gs://figgy-preservation
+        gsutil bucketpolicyonly set on gs://figgy-preservation
+        gsutil iam ch serviceAccount:figgy-preservation-production@pulibrary-figgy-storage-1.iam.gserviceaccount.com:objectAdmin gs://figgy-preservation
+        gsutil versioning set on gs://figgy-preservation
+        ```
+   1. When a resource is `complete` and marked with the `cloud` preservation
+      policy it will save itself and all resources contained in `member_ids` in
+      a directory structure in Google Cloud Storage that looks like the
+      following:
+      ```
+      - <resource-id>
+        - data
+          - <child-id>
+            - <child-id>.json
+            - <binary.tif>
+        - <resource-id>.json
+      ```
+   1. When a child with a preserved parent (any length up the hierarchy) is
```

Comment:
Can you reword this a bit? It's confusing whether the child or parent is completed, and which is updated.

Reply:
That better?